src/query/api/v1/httpd/handler.go

```diff
@@ -174,15 +174,15 @@ func (h *Handler) RegisterRoutes() error {
 		Path:               native.PromReadURL,
 		Handler:            h.options.QueryRouter(),
 		Methods:            native.PromReadHTTPMethods,
-		MiddlewareOverride: native.WithQueryParams,
+		MiddlewareOverride: native.WithQueryParamsAndRangeRewriting,
```

Comment:
Only apply the new middleware to prom handlers handling `/query` and `/query_range`